cmd/akash/deployment/create.go

```diff
@@ -0,0 +1,374 @@
+package deployment
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/gosuri/uitable"
+	"github.com/ovrclk/akash/cmd/akash/session"
+	"github.com/ovrclk/akash/cmd/common"
+	"github.com/ovrclk/akash/manifest"
+	"github.com/ovrclk/akash/marketplace"
+	"github.com/ovrclk/akash/provider/http"
+	"github.com/ovrclk/akash/sdl"
+	"github.com/ovrclk/akash/txutil"
+	"github.com/ovrclk/akash/types"
+	. "github.com/ovrclk/akash/util"
+	"github.com/spf13/cobra"
+	tmctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+type deployEvent uint
+
+const (
+	ttl = int64(5)
+
+	eventDeployBegin deployEvent = iota + 1
+	eventBroadcastBegin
+	eventBroadcastDone
+	eventReceiveOrdersBegin
+	eventReceiveOrder
+	eventReceiveOrdersDone
+	eventReceiveFulfillmentsBegin
+	eventReceiveFulfillment
+	eventReceiveFulfillmentsDone
+	eventReceiveLeaseBegin
+	eventReceiveLease
+	eventSendManifest
+	eventSendManifestDone
+	eventReceiveLeaseDone
+	eventDeployDone
+)
+
+type deployState struct {
+	fulfilments         []*types.TxCreateFulfillment
+	leases              []*types.TxCreateLease
+	providerLeaseStatus map[*types.Provider]*types.LeaseStatusResponse
+	groups              []*types.GroupSpec
+	orders              []*types.TxCreateOrder
+	mani                *types.Manifest
+	hash                []byte
+}
+
+var (
+	state = &deployState{
+		providerLeaseStatus: make(map[*types.Provider]*types.LeaseStatusResponse),
+	}
+	mtx sync.Mutex
+)
+
+type deployStatus struct {
+	Event   deployEvent
+	Message string
+	Error   error
+	Result  interface{}
+}
+
+func createCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create <deployment-file>",
+		Short: "create a deployment",
+		// Args:  cobra.ExactArgs(1),
+		RunE: session.WithSession(
+			session.RequireKey(session.RequireNode(create))),
+	}
+
+	session.AddFlagNode(cmd, cmd.Flags())
+	session.AddFlagKey(cmd, cmd.Flags())
+	session.AddFlagNonce(cmd, cmd.Flags())
+	session.AddFlagWait(cmd, cmd.Flags())
+	return cmd
+}
+
+func create(ses session.Session, cmd *cobra.Command, args []string) error {
+	var filePath string
+	if len(args) == 1 {
+		filePath = args[0]
+	}
+	filePath = ses.Mode().Ask().StringVar(filePath, "Deployment File Path (required): ", true)
+
+	statusChan := make(chan *deployStatus)
+	go func() {
```

Comment:
This is super cool but I don't really see why it has to be in a goroutine.  Can you print the stuff as it happens?  Maybe you're planning on making that event handling generic/pluggable?